packages/cli/src/Server.ts

```diff
@@ -887,12 +887,24 @@ class App {
 			const nodeTypes = NodeTypes();
 
 			const allNodes = nodeTypes.getAll();
+			const language = req.headers['accept-language'] ?? '';
+			const translations = await Promise.all(allNodes.map(async node => {
+				const nodeName = node.description.name.split('.')[1];
+				const {sourcePath} = nodeTypes.getByName(node.description.name, true);
+				const translationPath = path.join(path.dirname(sourcePath), 'translations', `${language}.${nodeName}.js`);
+				try {
+					const {[language]: {[nodeName]: {displayName, description}}} = await import(translationPath);
+					return {[language]: {[nodeName]: {displayName, description}}};
+				} catch (e) {
+					return undefined;
+				}
+			}));
 
-			allNodes.forEach((nodeData) => {
+			allNodes.forEach((nodeData, idx) => {
 				// Make a copy of the object. If we don't do this, then when
 				// The method below is called the properties are removed for good
 				// This happens because nodes are returned as reference.
-				const nodeInfo: INodeTypeDescription = { ...nodeData.description };
+				const nodeInfo: INodeTypeDescription = {...nodeData.description, translation: translations[idx]};
```

Comment:
Actually I also thought about this problem and therefore I take from translations and put into nodes only `displayName` and `description` so it doesn't seriously enlarge response from the server

Reply:
Ohhh...I think I got your point
Is that okay that we will make people to maintain multiple file to keep node translated?
It seems to me that in terms of the number of nodes there won't be lots of load to import the translation file on server

Reply:
y you are right that you just responded with these two keys, however you loaded and parsed the whole file from the filesystem, question is whether it takes significantly longer or not 🤔  You have translated most of the nodes so you might have a better picture about this. Accordingly open to take your suggestion and stick to the way you did it (without a `header` file).

Reply:
one thing that I forgot to mention is also the usage of the dynamic `import`. Since it is caching we would use memory for translations, however the browser of the user is caching as well. Thus, we should think about using `readFile` instead? What do you think?

Reply:
It is used as `import` only in TS syntax. On build it transforms to `await Promise.resolve().then(() => require(translationPath))`. But require still caches date afaik. Such approach can improve performance for all users who bootstrap n8n after it was bootstrapped for the first time.
As for memory usage avg size of translation file is 18kb and considering the number of nodes it gives us 1mb of consumed memory at max. Not crucial but can increase performance